Sources/TuistGenerator/Generator/ProjectGenerator.swift

```diff
@@ -248,12 +259,23 @@ final class ProjectGenerator: ProjectGenerating {
                                                    generatedProject: generatedProject)
     }
 
+    enum XcodeVersionError: Swift.Error {
+        case noXcode
+        case noVersion
+    }
+
     private func determineProjectConstants() -> ProjectConstants {
-        // TODO:
-        //
-        // To maintain backwards compatibility with Xcode 10
-        // the Xcode10 constants are used unless the use of Xcode 11
-        // features are detected (e.g Swift PM dependencies)
-        return .xcode10
+        do {
+            let version = try XcodeController(system: system).selectedVersion()
```

Comment:
Nitpick: I'd expose a shared instance from `XcodeController` that can be mocked easily as we already do with the printer or the file handler.

Reply:
Done 👍